tests/ert/unit_tests/storage/snapshots/test_storage_migration/test_that_storage_matches/responses

```diff
@@ -1 +1 @@
-{'gen_data': GenDataConfig(name='gen_data', input_files=['gen%d.txt'], keys=['GEN'], report_steps_list=[[1]]), 'summary': SummaryConfig(name='summary', input_files=['CASE'], keys=['FOPR', 'RWPR'], refcase={})}
+{'gen_data': GenDataConfig(name='gen_data', input_files=['gen%d.txt'], keys=['GEN'], has_finalized_keys=True, report_steps_list=[[1]]), 'summary': SummaryConfig(name='summary', input_files=['CASE'], keys=['FOPR', 'RWPR'], has_finalized_keys=False, refcase={})}
```

Comment:
In this case, `SummaryConfig` probably has finalized keys? Should we have some sort of logic for that? For example if there are no `*` in `keys` -> `has_finalized_keys` == True?

Reply:
Hmm, the way it is set up now, the `has_finalized_keys` is set initially in the response config implementation, so for `SummaryConfig` it always starts as `False`. Then when some responses are saved, those keys are internalized and the flag is switched to `True`. So the logic is there, but the flag is set explicitly when keys are saved explicitly. 

Corner case of having it like this is if the flag is defaulted to `True` and somehow we also have wildcards (would not work), or if we get different keys saved from different responses (should also not happen)

Reply:
Would keep the default to `False` I think, but perhaps have it changed in a `__post_init__`, but might be overkill.

Reply:
I think we can adjust it later and do something more in `__post_init__` if it becomes necessary, but for now I think we can have it defaulted to `False` in the `ResponseConfig` and `GenDataConfig`, and `True` in `SummaryConfig`, which is in line with the current behavior.